python_modules/dagster/dagster/_core/blueprints/load_from_yaml.py

```diff
@@ -121,3 +123,30 @@ def load_defs_from_yaml(
     ]
 
     return BlueprintDefinitions.merge(*def_sets_with_code_references).to_definitions()
+
+
+class YamlBlueprintsLoader(NamedTuple):
+    """A loader is responsible for loading a set of Dagster definitions from one or more YAML
+    files based on a set of supplied blueprints.
+    """
+
+    path: Union[Path, str]
```

Comment:
Even though I was the cause of this in the first place, I think better to stick to a single type - probably `Path` for the class attribute. And we can let the constructor accept a string and canonicalize it if we want.